ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/moreh_dot_device_operation.cpp

```diff
@@ -0,0 +1,88 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_dot_device_operation.hpp"
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/moreh_helper_functions.hpp"
+
+namespace ttnn::operations::moreh::moreh_dot_op {
+MorehDotOperation::program_factory_t MorehDotOperation::select_program_factory(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    // For now we litteraly don't care and return a single factory. Whatever
+    return SingleCore{};
+}
+
+void MorehDotOperation::validate(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    const auto& input_tensor_a = tensor_args.input_tensor_a;
+    const auto& input_tensor_b = tensor_args.input_tensor_b;
+
+    TT_ASSERT(tt::operations::primary::is_1d_tensor(input_tensor_a));
```

Comment:
Please see this comment
https://github.com/tenstorrent/tt-metal/pull/12296/files#r1756987856

Reply:
Replace with TT_FATAL and add an error description. 
Follow
See https://github.com/tenstorrent/tt-metal/blob/main/ErrorMessageBestPractices.md